src/TorchSharp/Tensor/Factories/tensor_float.cs

```diff
@@ -22,37 +22,22 @@ public static Tensor tensor(float scalar, Device? device = null, bool requires_g
         }
 
         /// <summary>
-        /// Create a scalar tensor from a single value
+        /// Create a scalar complex number tensor from independent real and imaginary components
         /// </summary>
         public static Tensor tensor(float real, float imaginary, ScalarType? dtype = null, Device? device = null, bool requires_grad = false)
         {
-            device = InitializeDevice(device);
-            var handle = THSTensor_newComplexFloat32Scalar(real, imaginary, (int)device.type, device.index, requires_grad);
-            if (handle == IntPtr.Zero) { CheckForErrors(); }
-            var tensor = new Tensor(handle);
-            if (device is { }) {
-                tensor = dtype.HasValue ? tensor.to(dtype.Value, device) : tensor.to(device);
-            } else if (dtype.HasValue) {
-                tensor = tensor.to_type(dtype.Value);
-            }
-            return tensor;
+            return tensor((real, imaginary), dtype: dtype, device: device);
         }
 
         /// <summary>
-        /// Create a scalar tensor from a single value
+        /// Create a scalar complex number tensor from a tuple of (real, imaginary)
         /// </summary>
         public static Tensor tensor((float Real, float Imaginary) scalar, ScalarType? dtype = null, Device? device = null, bool requires_grad = false)
         {
             device = InitializeDevice(device);
             var handle = THSTensor_newComplexFloat32Scalar(scalar.Real, scalar.Imaginary, (int)device.type, device.index, requires_grad);
             if (handle == IntPtr.Zero) { CheckForErrors(); }
-            var tensor = new Tensor(handle);
-            if (device is { }) {
-                tensor = dtype.HasValue ? tensor.to(dtype.Value, device) : tensor.to(device);
-            } else if (dtype.HasValue) {
-                tensor = tensor.to_type(dtype.Value);
-            }
-            return tensor;
+            return InstantiateTensorWithLeakSafeTypeChange(handle, dtype);
```

Comment:
See comments for double-precision complex numbers.

Reply:
Fixed.